optimus_manager/kernel.py

```diff
@@ -186,7 +185,8 @@ def _unload_bbswitch():
     try:
         exec_bash("modprobe -r bbswitch")
     except BashError as e:
-        raise KernelSetupError("Cannot unload bbswitch : %s" % str(e))
+        if not "Module bbswitch not found" in str(e):
```

Comment:
Why not use the already existing `is_module_loaded()` function to check if the bbswitch module is loaded before trying to unload it?

Reply:
I was thinking about using this function too. However, is_module_loaded() executes another system call and is therefore slower, see https://github.com/Askannz/optimus-manager/blob/6a74fd90287d7719354777e8bcd191516743bfa0/optimus_manager/checks.py#L34

This PR should be seen as a quick-fix only. Ideally, the whole _set_base_state() function should be improved / refactored to prevent the call to _unload_bbswitch on systems without bbswitch in the first place. I'm not aware of the conflicts which were resolved by unloading bbswitch at this place in the code (see https://github.com/Askannz/optimus-manager/commit/9da27f19a3987bbcd771a7e9805e75362a9e4615) as bbswitch doesn't work on my setup. Maybe @Askannz can help here?

Reply:
> I'm not aware of the conflicts which were resolved by unloading bbswitch at this place in the code
IIRC I added that because bbswitch goes into a panic if you try to reset the card when it's loaded. Hence the cycle "module load"->"power on"->"module unload"->"reset"->"module load" cycle. But the issue here is that I put the unload at a very stupid place, therefore it happens even if the "reset" feature is disabled. That's bug number 1.

Bug number 2 is that rmmod errors out if bbswitch isn't installed for the current kernel. Your solution for that should work but I'm not too keen on string-searching the error message to detect that particular case...as you mentioned, the proper fix would be to *not* send that command on systems where bbswitch isn't installed, but that's not possible without checking for its presence in some way. AFAIK `modinfo <module>` is the only reliable method for that, but if you are aware of a better one please let me know.

As a side note I tried adding to `-q` option to modprobe but that only hides the error message, not the non-zero return code :/